catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/PenBricksTest.java

```diff
@@ -0,0 +1,96 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.content.brick.app;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.bricks.PenDownBrick;
+import org.catrobat.catroid.content.bricks.PenUpBrick;
+import org.catrobat.catroid.content.bricks.SetPenColorBrick;
+import org.catrobat.catroid.content.bricks.SetPenSizeBrick;
+import org.catrobat.catroid.ui.ScriptActivity;
+import org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils;
+import org.catrobat.catroid.uiespresso.testsuites.Cat;
+import org.catrobat.catroid.uiespresso.testsuites.Level;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.action.ViewActions.click;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+
+@RunWith(AndroidJUnit4.class)
+public class PenBricksTest {
+	@Rule
+	public BaseActivityInstrumentationRule<ScriptActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(ScriptActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		createProject();
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testPenBricks() {
+		int penSize = 10;
+		int red = 1;
+		int green = 2;
+		int blue = 3;
+
+		onBrickAtPosition(0).checkShowsText(R.string.brick_when_started);
+		onBrickAtPosition(1).checkShowsText(R.string.brick_pen_down);
+		onBrickAtPosition(2).checkShowsText(R.string.brick_pen_up);
+		onBrickAtPosition(3).checkShowsText(R.string.brick_pen_size);
+		onBrickAtPosition(3).onFormulaTextField(R.id.brick_set_pen_size_edit_text).checkShowsNumber(penSize);
+		onBrickAtPosition(4).onFormulaTextField(R.id.brick_set_pen_color_action_red_edit_text).checkShowsNumber(red);
+		onBrickAtPosition(4).onFormulaTextField(R.id.brick_set_pen_color_action_green_edit_text).checkShowsNumber(green);
+		onBrickAtPosition(4).onFormulaTextField(R.id.brick_set_pen_color_action_blue_edit_text).checkShowsNumber(blue);
+		onBrickAtPosition(4).onFormulaTextField(R.id.brick_set_pen_color_action_blue_edit_text).perform(click());
```

Comment:
why is the last one doing a click, while all others are just checked?

Reply:
similar bricks with formula edit fields at least test if a click opens the formula editor, and if a entered formula is also correctly displayed on the brick afterwards.